aioredisqueue/task.py

```diff
@@ -12,13 +13,21 @@ def generate_ack_info():
         return int(timeit.default_timer() * 1000)
 
     @staticmethod
-    def generate_id(payload=None):
-        return str(uuid.uuid4())
+    def generate_id(serialization_method):
+        return str(uuid.uuid4()) + '_' + serialization_method
 
     @staticmethod
-    def format_payload(payload):
-        return payload
-
+    def encode_payload(payload, serialization_method):
+        dump = get_serializer(serialization_method)['dump']
+        return dump(payload)
+    
+    @staticmethod
+    def decode_payload(task_id, payload):
+        task_id = task_id.decode('utf-8')
+        serialization_method = task_id[task_id.find('_') + 1:]
```

Comment:
what if task_id doesn't contain `_` and `serialization method`? (current behavior)

Reply:
Current behavior is opposite, `task_id` does contian `_` and `serialization method`. It is guaranteed by `Task.generate_id()`

Reply:
Currently in `master` we have:

```python
    @staticmethod
    def generate_id(payload=None):
        return str(uuid.uuid4())

Reply:
This function is intended to be merged into `master` with following version of `generate_id` :
```python
    def generate_id(serialization_method):
        return str(uuid.uuid4()) + '_' + serialization_method
```

Reply:
So to update queue to new version, the tasks must be drained? That should be fixed or at least documented as backward-incompatible change.